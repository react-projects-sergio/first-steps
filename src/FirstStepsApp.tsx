import { ItemCounter } from "./shopping-cart/ItemCounter";

interface ItemInCart {
  productName: string;
  quantity: number;
}

const itemsInCart: ItemInCart[] = [
  { productName: "Nintendo Switch 2", quantity: 1 },
  { productName: "Pro Controller", quantity: 2 },
  { productName: "Mario Kart World", quantity: 5 },
];

export function FirstStepsApp() {
  return (
    <>
      <h1>Carrito de compras</h1>
      {itemsInCart.map(({ productName, quantity }) => (
        <ItemCounter key={productName} name={productName} quantity={quantity} />
      ))}
    </>
  );
}
