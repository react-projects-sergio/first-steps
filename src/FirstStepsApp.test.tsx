import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { FirstStepsApp } from "./FirstStepsApp";

describe("FirstStepsApp", () => {
  test("should match snapshot", () => {
    const { container } = render(<FirstStepsApp />);

    expect(container).toMatchSnapshot();
  });
});
