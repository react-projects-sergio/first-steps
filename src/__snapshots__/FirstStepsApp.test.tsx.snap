// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp > should match snapshot 1`] = `
<div>
  <h1>
    Carrito de compras
  </h1>
  <section
    class="_item-row_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: red;"
    >
      Nintendo Switch 2
    </span>
    <button>
      +1
    </button>
    <span>
      1
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_item-row_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Pro Controller
    </span>
    <button>
      +1
    </button>
    <span>
      2
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_item-row_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Mario Kart World
    </span>
    <button>
      +1
    </button>
    <span>
      5
    </span>
    <button>
      -1
    </button>
  </section>
</div>
`;
